alacritty/src/display/mod.rs

```diff
@@ -188,9 +188,17 @@ pub struct Display {
 
 impl Display {
     pub fn new<E>(config: &Config, event_loop: &EventLoop<E>) -> Result<Display, Error> {
-        // Guess DPR based on first monitor.
-        let estimated_dpr =
-            event_loop.available_monitors().next().map(|m| m.scale_factor()).unwrap_or(1.);
+        #[cfg(any(not(feature = "x11"), target_os = "macos", windows))]
+        let is_x11 = false;
+        #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+        let is_x11 = event_loop.is_x11();
+
+        // Guess DPR based on first monitor, except on Wayland.
```

Comment:
I'd love to get a little more comment on why we don't do this on Wayland, since I'm certain I'll forget about it in about a week or two.

```suggestion
        // Guess DPR based on first monitor. On Wayland the initial frame always renders at a DPR of 1.
```

Reply:
Smart.